core/src/components/segment/segment.tsx

```diff
@@ -431,6 +442,83 @@ export class Segment implements ComponentInterface {
     this.checked = current;
   }
 
+  private getSegmentButton = (selector: 'first' | 'last' | 'next' | 'previous' | 'focused'): HTMLIonSegmentButtonElement | null => {
+    const buttons = this.getButtons().filter(button => !button.disabled);
+
+    /* When `selectOnFocus` is true, the current button is the checked button.
+     * Otherwise, it is the button that is currently focused.
+    **/
+    const currIndex = buttons.findIndex(button => {
+      if (this.selectOnFocus && this.checked) {
+        return button === this.checked;
+      }
+      return button === document.activeElement;
+    });
+
+    switch (selector) {
+      case 'first':
+        return buttons[0];
+      case 'last':
+        return buttons[buttons.length - 1];
+      case 'next':
+        return buttons[currIndex + 1] || buttons[0];
+      case 'previous':
+        return buttons[currIndex - 1 ] || buttons[buttons.length - 1];
+      case 'focused':
+        return this.el.querySelector('ion-segment-button.ion-focused');
+      default:
+        return null;
+    }
+  }
+
+  @Listen('keydown')
+  onKeyDown(ev: KeyboardEvent) {
+    const isRTL = document.dir === 'rtl';
+    let select = this.selectOnFocus;
+    let current;
+    switch (ev.key) {
+      case 'ArrowRight':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('previous') : this.getSegmentButton('next');
+        break;
+      case 'ArrowLeft':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('next') : this.getSegmentButton('previous')
+        break;
+      case 'Home':
+        ev.preventDefault();
+        current = this.getSegmentButton('first');
+        break;
+      case 'End':
+        ev.preventDefault();
+        current = this.getSegmentButton('last');
+        break;
+      case ' ':
+      case 'Enter':
+        ev.preventDefault();
+        current = this.getSegmentButton('focused');
```

Comment:
```suggestion
        current = document.activeElement;
```

If an `ion-segment-button` is focused then it will be `document.activeElement`. If segment is not focused then this keydown handler will not fire anyways since it is attached to `ion-segment`.